geode-web-api/src/main/java/org/apache/geode/rest/internal/web/controllers/CommonCrudController.java

```diff
@@ -73,7 +75,12 @@
     logger.debug("Listing all resources (Regions) in Geode...");
     final HttpHeaders headers = new HttpHeaders();
     headers.setLocation(toUri());
-    final Set<Region<?, ?>> regions = getCache().rootRegions();
+    final Set<Region<?, ?>> regions = new HashSet<>();
+    for (InternalRegion region : getCache().getApplicationRegions()) {
+      if (region instanceof Region) {
```

Comment:
InternalRegion extends Region, do we need instance check.

Reply:
The java compiler would not let me add the InternalRegions to the "regions" HashSet without casting to Region first.  Since that was required I felt obliged to perform an instanceof check.